dash/dash.py

```diff
@@ -565,4 +565,13 @@ def run_server(self,
                    port=8050,
                    debug=False,
                    **flask_run_options):
+        # Make sure `layout` is set before running the server
+        value = getattr(self, 'layout')
+        if value is None:
+            raise exceptions.NoLayoutException(
+                ''
+                'The layout was `None` '
+                'at the time that `run_server` was called. '
+                'Make sure to set the `layout` attribute of your application '
+                'before running the server.')
```

Comment:
@ned2 Brought a good point in #286, anything in `run_server` won't be run by wsgi since it's called in `if __name__ == '__main__':`. Should move that check to `_setup_server`.

Reply:
good point, looks like it will also work in `_setup_server`